src/lib/histogram.lua

```diff
@@ -213,8 +259,25 @@ function wrap_thunk(histogram, thunk, now)
    end
 end
 
+function iterate(histogram)
+   print("histogram is", histogram, histogram.growth_factor_log)
+   local lo = histogram.minimum
+   local count = histogram.count
+   local hi = lo * (math.exp(histogram.growth_factor_log)^(num_buckets - 2))
+   local entry = 0
+   local max_entry = num_buckets
+   local function next_entry()
+      if entry < max_entry then
+         entry = entry + 1
+         return histogram.buckets[entry - 1], lo, hi, count
+      end
+   end
+   return next_entry
+end
```

Comment:
So, I know I've said this like 3 times, but apologies again for the miscommunication!  In https://github.com/SnabbCo/snabbswitch/pull/795 I was working on upstreaming the histogram in parallel, ended up having to refactor, and as I wasn't aware of your work I implemented the same thing.

Would you mind reviewing the iterate() function there?  It's essentially the same, only that it returns the bucket index but not the total when itrerating, and can take a "prev" argument.